lib/registry.dart

```diff
@@ -281,8 +281,8 @@ class RegistryManager {
 
   /// Maps id to entry. Each entry contains the id, the element, its tags,
   /// and a capability required to remove it again.
-  Map<int, _RegistryEntry> _entries = new HashMap();
-  Map<Object, Set<int>> _tag2id = new HashMap();
+  final Map<int, _RegistryEntry> _entries = new HashMap();
```

Comment:
Are we sure this is inferred correctly on the right-hand side?

Reply:
Gah! I wished I'd put the type on the right. Inference will take care of this, though...

Reply:
Putting the type arguments on the right would *not* work. The variable type is `Map` and the constructed object is `HashMap`, so just putting `var` on the left is wrong. Putting `Map` alone makes it `Map<dynamic,dynamic>`, so you really *need* the full type on the left.
(I just couldn't tell whether the arguments are correctly inferred for `HashMap` when the left-hand side says `Map<...>`, good to know that that works).